lib/framework/SystemStatus.cpp

```
@@ -169,6 +174,13 @@ esp_err_t SystemStatus::systemStatus(PsychicRequest *request)
     root["core_temp"] = temperatureRead();
     root["cpu_reset_reason"] = verbosePrintResetReason(esp_reset_reason());
     root["uptime"] = millis() / 1000;
+    #ifdef CONFIG_IDF_TARGET_ESP32P4
+        esp_hosted_coprocessor_fwver_t c6_fw_version;
+        esp_hosted_get_coprocessor_fwversion(&c6_fw_version);
+        char hfw[10];
+        snprintf(hfw, 32, "v%d.%d.%d", c6_fw_version.major1, c6_fw_version.minor1, c6_fw_version.patch1);
```

Cppcheck (2.18.0)

[error] Buffer is accessed out of bounds

(bufferAccessOutOfBounds)